src/meshes/p4est_mesh.jl

```diff
@@ -517,6 +572,242 @@ function connectivity_structured(n_cells_x, n_cells_y, n_cells_z, periodicity)
+      elseif direction == 4
+        target = 5
+        tree_to_tree[3, tree] = linear_indices[cell_x, end, cell_z, target] - 1
+        tree_to_face[3, tree] = 3
+      elseif direction == 5
+        target = 3
+        tree_to_tree[3, tree] = linear_indices[end - cell_x + 1, 1, cell_z, target] - 1
+        tree_to_face[3, tree] = 8 # first face dimensions are oppositely oriented, add 6
+      else # direction == 6
+        target = 3
+        tree_to_tree[3, tree] = linear_indices[cell_x, end, cell_z, target] - 1
+        tree_to_face[3, tree] = 3
+      end
+
+      # Positive y-direction
+      if cell_y < n_cells_y # Connect to tree at the same face
+        tree_to_tree[4, tree] = linear_indices[cell_x, cell_y + 1, cell_z, direction] - 1
+        tree_to_face[4, tree] = 2
+      elseif direction == 1
+        target = 6
+        tree_to_tree[4, tree] = linear_indices[1, end - cell_x + 1, cell_z, target] - 1
+        tree_to_face[4, tree] = 6 # first face dimensions are oppositely oriented, add 6
+      elseif direction == 2
+        target = 6
+        tree_to_tree[4, tree] = linear_indices[end, cell_x, cell_z, target] - 1
+        tree_to_face[4, tree] = 1
+      elseif direction == 3
+        target = 6
+        tree_to_tree[4, tree] = linear_indices[cell_x, 1, cell_z, target] - 1
+        tree_to_face[4, tree] = 2
+      elseif direction == 4
+        target = 6
+        tree_to_tree[4, tree] = linear_indices[end - cell_x + 1, end, cell_z, target] - 1
+        tree_to_face[4, tree] = 9 # first face dimensions are oppositely oriented, add 6
+      elseif direction == 5
+        target = 4
+        tree_to_tree[4, tree] = linear_indices[cell_x, 1, cell_z, target] - 1
+        tree_to_face[4, tree] = 2
+      else # direction == 6
+        target = 4
+        tree_to_tree[4, tree] = linear_indices[end - cell_x + 1, end, cell_z, target] - 1
+        tree_to_face[4, tree] = 9 # first face dimensions are oppositely oriented, add 6
+      end
+
+      # Negative z-direction
+      if cell_z > 1
+        tree_to_tree[5, tree] = linear_indices[cell_x, cell_y, cell_z - 1, direction] - 1
+        tree_to_face[5, tree] = 5
+      else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+        tree_to_tree[5, tree] = tree - 1
+        tree_to_face[5, tree] = 4
+      end
+
+      # Positive z-direction
+      if cell_z < n_cells_z
+        tree_to_tree[6, tree] = linear_indices[cell_x, cell_y, cell_z + 1, direction] - 1
+        tree_to_face[6, tree] = 4
+      else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+        tree_to_tree[6, tree] = tree - 1
+        tree_to_face[6, tree] = 5
+      end
+    end
+  end
+
+  tree_to_edge = C_NULL
+  # p4est docs: "in trivial cases it is just a pointer to a p4est_topix value of 0."
+  # We don't need edge connectivity, so this is a trivial case.
+  ett_offset = Array{p4est_topidx_t}([0])
```

Comment:
```suggestion
  ett_offset = zeros(p4est_topidx_t, 1)
```
Maybe this is a slightly clearer syntax (also in similar cases later)?